src/Fixer/ReturnNotation/SimplifiedNullReturnFixer.php

```diff
@@ -137,13 +140,34 @@ private function isStrictOrNullableReturnTypeFunction(Tokens $tokens, int $retur
     /**
      * Should we clear the specific token?
      *
-     * If the token is a comment, or is whitespace that is immediately before a
-     * comment, then we'll leave it alone.
+     * We'll leave it alone if
+     * - token is a comment
+     * - token is whitespace that is immediately before a comment
+     * - token is whitespace that is immediately before the PHP close tag
+     * - token is whitespace that is immediately after a comment and before a semicolon
      */
     private function shouldClearToken(Tokens $tokens, int $index): bool
     {
         $token = $tokens[$index];
 
-        return !$token->isComment() && !($token->isWhitespace() && $tokens[$index + 1]->isComment());
+        if ($token->isComment()) {
+            return false;
+        }
+
+        if ($token->isWhitespace()) {
```

Comment:
We can avoid nesting if by doing `if (!$token->isWhitespace()) { return true; }`, right?

And then, maybe, a single return with all 3 nested conditions?